docs/documenting/yaml-style-guide.md

```diff
@@ -0,0 +1,483 @@
+---
+title: "YAML Style Guide"
+---
+
+Additional to our general [documentation standards](/documenting/standards.md),
+also have a set of standards for documenting snippets of YAML. These standards
+described on this page, apply to our all YAML based code across the project,
+with the main focus on documentation.
+
+Our YAML standards provide our end-users with a consistent look, the best
+practices and a uniform approach for solving problems in YAML.
+
+## YAML
+
+This paragraph is about basic YAML usage, and thus not Home Assistant specific.
+
+### Indentation
+
+An indentation of 2 spaces must be used.
+
+```yaml
+# Good
+example:
+  one: 1
+
+# Bad
+example:
+    bad: 2
+```
+
+### Booleans
+
+We should avoid the use of truthy boolean values in YAML. They often trow
+of people new to YAML. Therefore, we only allow the use of `true` and `false`
+as boolean values, in lower case.
+
+This keeps it compatible with the YAML 1.2 specifications as well, since that
+version dropped support for several unquotes truthy booleans (e.g., `y`, `n`,
+`yes`, `no`, `on`, `off` and similar).
+
+  ```yaml
+# Good
+one: true
+two: false
+
+# Bad
+one: True
+two: on
+three: yes
+```
+
+### Comments
+
+Adding comments to blocks of YAML, can really help the reader understand the
+example better.
+
+The indentation level of the comment must match the current indentation level. Preferable is written above the line the comment applies to, otherwise lines
```

Comment:
```suggestion
The indentation level of the comment must match the current indentation level. Preferably the comment is written above the line the comment applies to, otherwise lines
```